### PR TITLE
Migrate dotnet package search to use system.commandLine

### PR DESCRIPTION
**Pull request by Nigusu-Allehu**
## Bug

Fixes: https://github.com/NuGet/Home/issues/13031

Regression? Last working version:

## Description

This PR migrates  `dotnet package search` command in `NuGet.CommandLine.XPlat` to use the `System.CommandLine` package while preserving the other commands in `NuGet.CommandLine.XPlat` to keep using the `Microsoft.Extensions.CommandLineUtils` library. The other commands will be migrated as part of the migration work @martinrrm is doing : https://github.com/NuGet/NuGet.Client/pull/5474 .

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by nkolev92 — APPROVED** [review]

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs [hunk 1]
```diff
@@ -2,84 +2,114 @@
...
 
-        public static void Register(CommandLineApplication app, Func<ILoggerWithColor> getLogger, Func<PackageSearchArgs, Task<int>> setupSettingsAndRunSearchAsync)
+        public static void Register(CliCommand rootCommand, Func<ILoggerWithColor> getLogger, Func<PackageSearchArgs, CancellationToken, Task<int>> setupSettingsAndRunSearchAsync)
         {
-            app.Command("search", pkgSearch =>
+            var searchCommand = new CliCommand("search", Strings.pkgSearch_Description);
+
+            var searchTerm = new CliArgument<string>("Search Term")
+            {
+                Description = Strings.pkgSearch_Description,
```
Comment (jgonz120) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs]: Should this be pkgSearch_termDescription?

**Review by jgonz120 — APPROVED** [review]

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs (hunk 1 above)
Comment (Nigusu-Allehu) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs]: Nice catch! Fixed.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs [hunk 2]
```diff
@@ -2,84 +2,113 @@
...
-
-                pkgSearch.OnExecute(async () =>
+                Description = Strings.pkgSearch_InteractiveDescription,
+                Arity = ArgumentArity.Zero
+            };
+
+            var help = new HelpOption("--help", "-h", "-?")
+            {
+                Arity = ArgumentArity.Zero
+            };
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs]: Is this needed? I thought System.CommandLine automatically adds the help option (at least `--help` and `-h`)

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs [hunk 3]
```diff
@@ -69,6 +73,51 @@ public static int MainInternal(string[] args, CommandOutputLogger log)
...
+                    return log;
+                };
+
+                CliCommand rootCommand = new CliCommand("package");
+                PackageSearchCommand.Register(rootCommand, getHidePrefixLogger);
+
+                args[0] = null;
+                args = args
+                    .Where(e => e != null)
+                    .ToArray();
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs]: args = args.Skip(1).ToArray();

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs [hunk 4]
```diff
@@ -69,6 +73,51 @@ public static int MainInternal(string[] args, CommandOutputLogger log)
...
+
+                foreach (CliCommand command in rootCommand.Subcommands)
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs]: Does `await rootCommand.InvokeAsync(args)` (or something like that) make this loop redundant?

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs (hunk 2 above)
Comment (Nigusu-Allehu) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs]: fixed

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs (hunk 4 above)
Comment (Nigusu-Allehu) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs]: fixed

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs [hunk 5]
```diff
@@ -2,84 +2,113 @@
...
+            var searchTerm = new CliArgument<string>("Search Term")
+            {
+                Description = Strings.pkgSearch_termDescription,
+                Arity = ArgumentArity.ZeroOrOne,
+            };
+
+            var sources = new CliOption<List<string>>("--source")
+            {
+                Description = Strings.pkgSearch_SourceDescription,
+                Arity = ArgumentArity.ZeroOrMore
```
Comment (donnie-msft) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs]:
```suggestion
                Arity = ArgumentArity.OneOrMore
```

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs [hunk 6]
```diff
@@ -2,84 +2,113 @@
...
-                    "--skip",
-                    Strings.pkgSearch_SkipDescription,
-                    CommandOptionType.SingleValue);
-
-                pkgSearch.OnExecute(async () =>
+                Description = Strings.pkgSearch_InteractiveDescription,
+                Arity = ArgumentArity.Zero
+            };
+
+            var help = new HelpOption()
```
Comment (donnie-msft) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs]: nit: If you move this one after `--skip`, the Add methods below will be in the same order.

**Pull request merged**